lib/ruby_lsp/requests/support/uri.rb

```diff
@@ -0,0 +1,45 @@
+# typed: strict
+# frozen_string_literal: true
+
+module URI
+  class Generic
+    class << self
+      extend T::Sig
+
+      sig { params(path: String, scheme: String).returns(URI::Generic) }
+      def from_path(path:, scheme: "file")
+        # On Windows, if the path begins with the disk name, we need to add a leading slash to make it a valid URI
+        escaped_path = if /^[A-Z]:/.match?(path)
+          DEFAULT_PARSER.escape("/#{path}")
+        else
+          DEFAULT_PARSER.escape(path)
+        end
+
+        build(scheme: scheme, path: escaped_path)
+      end
+    end
+
+    extend T::Sig
+
+    sig { returns(T.nilable(String)) }
+    def to_path
```

Comment:
I feel this name could be easily confused with the native `path` method and doesn't convey its purpose well. How about `unescaped_path`?